lnwallet/fee_estimator.go

```diff
@@ -200,3 +200,127 @@ func (b *BtcdFeeEstimator) fetchEstimatePerByte(confTarget uint32) (btcutil.Amou
+// with the bitcoind node, and also a fall back fee rate. The fallback fee rate
+// is used in the occasion that the estimator has insufficient data, or returns
+// zero for a fee estimate.
+func NewBitcoindFeeEstimator(rpcConfig rpcclient.ConnConfig,
+	fallBackFeeRate btcutil.Amount) (*BitcoindFeeEstimator, error) {
+
+	rpcConfig.DisableConnectOnNew = true
+	rpcConfig.DisableAutoReconnect = false
+	rpcConfig.DisableTLS = true
+	rpcConfig.HTTPPostMode = true
+	chainConn, err := rpcclient.New(&rpcConfig, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &BitcoindFeeEstimator{
+		fallBackFeeRate: fallBackFeeRate,
+		bitcoindConn:    chainConn,
+	}, nil
+}
+
+// Start signals the FeeEstimator to start any processes or goroutines
+// it needs to perform its duty.
+//
+// NOTE: This method is part of the FeeEstimator interface.
+func (b *BitcoindFeeEstimator) Start() error {
+	return nil
+}
+
+// Stop stops any spawned goroutines and cleans up the resources used
+// by the fee estimator.
+//
+// NOTE: This method is part of the FeeEstimator interface.
+func (b *BitcoindFeeEstimator) Stop() error {
+	return nil
+}
+
+// EstimateFeePerByte takes in a target for the number of blocks until an
+// initial confirmation and returns the estimated fee expressed in
+// satoshis/byte.
+func (b *BitcoindFeeEstimator) EstimateFeePerByte(numBlocks uint32) (btcutil.Amount, error) {
+	feeEstimate, err := b.fetchEstimatePerByte(numBlocks)
+	switch {
+	// If the estimator doesn't have enough data, or returns an error, then
+	// to return a proper value, then we'll return the default fall back
+	// fee rate.
+	case err != nil:
+		walletLog.Errorf("unable to query estimator: %v", err)
+		fallthrough
+
+	case feeEstimate == 0:
+		return b.fallBackFeeRate, nil
+	}
+
+	return feeEstimate, nil
+}
+
+// EstimateFeePerWeight takes in a target for the number of blocks until an
+// initial confirmation and returns the estimated fee expressed in
+// satoshis/weight.
+func (b *BitcoindFeeEstimator) EstimateFeePerWeight(numBlocks uint32) (btcutil.Amount, error) {
+	feePerByte, err := b.EstimateFeePerByte(numBlocks)
+	if err != nil {
+		return 0, err
+	}
+
+	// We'll scale down the fee per byte to fee per weight, as for each raw
+	// byte, there's 1/4 unit of weight mapped to it.
+	satWeight := feePerByte / blockchain.WitnessScaleFactor
+
+	// If this ends up scaling down to a zero sat/weight amount, then we'll
+	// use the default fallback fee rate.
+	if satWeight == 0 {
+		return b.fallBackFeeRate / blockchain.WitnessScaleFactor, nil
+	}
+
+	return satWeight, nil
+}
+
+// fetchEstimate returns a fee estimate for a transaction be be confirmed in
+// confTarget blocks. The estimate is returned in sat/byte.
+func (b *BitcoindFeeEstimator) fetchEstimatePerByte(confTarget uint32) (btcutil.Amount, error) {
+	// First, we'll fetch the estimate for our confirmation target.
+	btcPerKB, err := b.bitcoindConn.EstimateFee(int64(confTarget))
```

Comment:
I think we might want to use `estimatesmartfee` here. As it uses the updated fee estimation algorithms by Morcos.

Reply:
Thanks for that pointer! I hadn't seen it before and will update this call.

Reply:
This bit is now fixed.